src/types.rs

```diff
@@ -697,15 +740,29 @@ impl Rewrite for ast::TraitRef {
 
 impl Rewrite for ast::Ty {
     fn rewrite(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
+        self.rewrite_result(context, shape).ok()
+    }
+
+    fn rewrite_result(&self, context: &RewriteContext<'_>, shape: Shape) -> RewriteResult {
         match self.kind {
             ast::TyKind::TraitObject(ref bounds, tobj_syntax) => {
                 // we have to consider 'dyn' keyword is used or not!!!
                 let (shape, prefix) = match tobj_syntax {
-                    ast::TraitObjectSyntax::Dyn => (shape.offset_left(4)?, "dyn "),
-                    ast::TraitObjectSyntax::DynStar => (shape.offset_left(5)?, "dyn* "),
+                    ast::TraitObjectSyntax::Dyn => (
+                        shape
+                            .offset_left(4)
+                            .max_width_error(shape.width, self.span())?,
+                        "dyn ",
+                    ),
+                    ast::TraitObjectSyntax::DynStar => (
+                        shape
+                            .offset_left(5)
+                            .max_width_error(shape.width, self.span())?,
+                        "dyn* ",
+                    ),
```

Comment:
Just a minor nit, but I think the formatting might be nicer if we defined `shape` as a new variable. You might need to double check the formatting.

```suggestion
                    ast::TraitObjectSyntax::Dyn => {
                        let shape = shape.offset_left(4)
                            .max_width_error(shape.width, self.span())?;
                        (shape, "dyn ")
                    },
                    ast::TraitObjectSyntax::DynStar => {
                        let shape = shape.offset_left(5)
                            .max_width_error(shape.width, self.span())?;
                        (shape, "dyn* ")
                    },
```

Reply:
This looks nicer since we can know which one corresponds to (shape, prefix) at a glance. I pushed the change :)